### PR TITLE
Nudge emails

### PR DESCRIPTION
**Pull request by akssinhaamit**
# Motivation and Context

# What has changed

# How to test?

# Links

# Screenshots (if appropriate):

### CODE REVIEW

File: response_operations_ui/views/update_event_date.py [hunk 1]
```diff
@@ -89,11 +107,12 @@ def update_event_date_submit(short_name, period, tag):
     """Attempts to create the event, returns None if success or returns an error message upon failure."""
     error_message = collection_exercise_controllers.update_event(
         collection_exercise_id=exercise['id'], tag=tag, timestamp=submitted_dt)
-
     if error_message:
         flash(error_message, 'error')
         return redirect(url_for('collection_exercise_bp.update_event_date',
                                 short_name=short_name, period=period, tag=tag))
-
+    success_message = 'Event date updated.' if tag != 'return_by' else 'Event date updated. Nudge emails that were ' \
```
Comment (insacuri) [inline response_operations_ui/views/update_event_date.py:114]: This is kinda ugly.  Probably better to have this either be a long form if/else 
```python
# might have to do success_message = "" here (I forget python variable scoping.. ^^)
if tag != 'return_by':
    success_message = 'Event date updated.' 
else:
    success_message = 'Event date updated. Nudge emails that were scheduled after the return by date will be ' \
                                        'removed. 
    return redirect(url_for('collection_exercise_bp.view_collection_exercise',
                             short_name=short_name, period=period, success_panel=success_message))
```

or do something like 
```python
if tag != 'return_by':
    success_message = 'Event date updated.' 
    return redirect(url_for('collection_exercise_bp.view_collection_exercise',
                             short_name=short_name, period=period, success_panel=success_message))
else:
    success_message = 'Event date updated. Nudge emails that were scheduled after the return by date will be ' \
                                        'removed. 
    return redirect(url_for('collection_exercise_bp.view_collection_exercise',
                             short_name=short_name, period=period, success_panel=success_message))
```

File: tests/views/test_collection_exercise.py [hunk 2]
```diff
@@ -536,7 +540,7 @@ def test_view_collection_instrument(self, mock_request, mock_details):
     @patch('response_operations_ui.views.collection_exercise.build_collection_exercise_details')
     def test_choose_collection_instrument_when_first(self, mock_request, mock_details):
         with open(
-            "tests/test_data/collection_exercise/formatted_collection_exercise_details_no_ci.json"
+                "tests/test_data/collection_exercise/formatted_collection_exercise_details_no_ci.json"
```
Comment (insacuri) [inline tests/views/test_collection_exercise.py:543]: What's with the odd spacing?  It'd be better to have it all on 1 line.  If can, change this for all of them that you've changed in this file.

File: tests/views/test_collection_exercise.py [hunk 3]
```diff
@@ -1257,17 +1261,121 @@ def test_create_collection_events_not_set_sequentially(self, mock_request, mock_
...
-
             ce_list = get_collection_exercises_with_events_and_samples_by_survey_id(self.survey['id'])
 
         expected_ce_list = copy.deepcopy(self.collection_exercises)
         expected_ce_list[0]['events'] = self.collection_exercise_events
         expected_ce_list[0]['sample_summary'] = self.sample_summary
         self.assertEqual(ce_list, expected_ce_list)
+
+    @requests_mock.mock()
+    def test_schedule_nude_email_option_not_present(self, mock_request):
```
Comment (insacuri) [inline tests/views/test_collection_exercise.py:1279]: :%s/nude/nudge/g ^^

EDIT:  Fix all of them in this file.

File: response_operations_ui/controllers/collection_exercise_controllers.py [hunk 4]
```diff
@@ -82,6 +82,19 @@ def update_event(collection_exercise_id, tag, timestamp):
...
+def delete_event(collection_exercise_id, tag):
+    logger.info('Deleting collection exercise event', collection_exercise_id=collection_exercise_id, tag=tag)
+
+    url = f'{app.config["COLLECTION_EXERCISE_URL"]}/collectionexercises/{collection_exercise_id}/events/{tag}'
+    response = requests.Session().post(url=url, auth=app.config['BASIC_AUTH'])
+
+    response.raise_for_status()
+
+    logger.info('Successfully deleted event', collection_exercise_id=collection_exercise_id,
+                tag=tag, )
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply response_operations_ui/controllers/collection_exercise_controllers.py:94]: done!

File: response_operations_ui/forms.py [hunk 5]
```diff
@@ -4,8 +4,8 @@
 
 from flask_wtf import FlaskForm
 from structlog import wrap_logger
-from wtforms import BooleanField, HiddenField, IntegerField, Label, PasswordField, SelectField, StringField, \
-    SubmitField, TextAreaField
+from wtforms import HiddenField, IntegerField, Label, PasswordField, SelectField, StringField, \
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply response_operations_ui/forms.py]: it was git merge ... brought back to the alphabetic order

File: response_operations_ui/templates/update-event-date.html [hunk 6]
```diff
@@ -86,7 +86,20 @@ <h2 id='error-saving-message' class="venus">Error updating {{event_name}} date <
...
-
+      {% if 'nudge' in event_name %}
+        <div class="u-mb-s">
+          <fieldset class="fieldset">
+              <div class="checkboxes__items">
+                  <p class="checkboxes__item">
+                  <span class="checkbox ">
+                {{ form.checkbox(class="checkbox__input js-checkbox ") }}
+                <label class="checkbox__label">Remove Event</label>
+              </span>
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply response_operations_ui/templates/update-event-date.html]: done!

File: response_operations_ui/views/update_event_date.py (hunk 1 above)
Comment (akssinhaamit) [reply response_operations_ui/views/update_event_date.py]: ok done!

File: tests/views/test_collection_exercise.py (hunk 2 above)
Comment (akssinhaamit) [reply tests/views/test_collection_exercise.py]: This is the ESlint, don't think its a harm looks good.

File: tests/views/test_collection_exercise.py (hunk 3 above)
Comment (akssinhaamit) [reply tests/views/test_collection_exercise.py]: fixed

**Pull request merged**